### PR TITLE
Optimize code to remove phpmd suppress warnings

### PR DESCRIPTION
**Pull request by arnoudhgz**
### Description
By making the order of some checks more efficient the suppress warnings in these two files about CyclomaticComplexity and NPathComplexity.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @arnoudhgz. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Customer/Model/Metadata/Form/Text.php [hunk 1]
```diff
@@ -9,34 +9,40 @@
...
+     * @param TimezoneInterface $localeDate
+     * @param LoggerInterface $logger
      * @param AttributeMetadataInterface $attribute
-     * @param \Magento\Framework\Locale\ResolverInterface $localeResolver
+     * @param ResolverInterface $localeResolver
      * @param string $value
      * @param string $entityTypeCode
      * @param bool $isAjax
-     * @param \Magento\Framework\Stdlib\StringUtils $stringHelper
+     * @param StringUtils $stringHelper
```
Comment (orlangur) [inline app/code/Magento/Customer/Model/Metadata/Form/Text.php]: It's better to avoid such changes, they should be done as separate efforts.

**Comment by arnoudhgz** [comment]
@orlangur I updated the PR to only solve the suppress warnings issues, the import statements have been undone.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @arnoudhgz,
In general - looks good, but there are some tests that are failing. Could you look into it?

**Comment by ihor-sviziev** [comment]
Hi @arnoudhgz,
Will you be able to look at failing tests? Maybe you need some help? Of so - you can contact me at Slack

**Comment by arnoudhgz** [comment]
@ihor-sviziev I think later this week I will look into it. Have been busy for a while.

**Comment by arnoudhgz** [comment]
@ihor-sviziev I used a wrong method, it is solved now

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-3326](https://jira.corp.magento.com/browse/ENGCOM-3326) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @arnoudhgz. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**